src/shared/specifier.c

```diff
@@ -208,6 +208,21 @@ int specifier_short_host_name(char specifier, const void *data, const char *root
         return 0;
 }
 
+int specifier_pretty_host_name(char specifier, const void *data, const char *root, const void *userdata, char **ret) {
+        char *n = NULL;
+        int r;
+
```

Comment:
`assert(ret)`

Reply:
Given that almost none of the specifier helper functions assert on `ret`, it's been either left out intentionally or should be added everywhere :-)